x-pack/plugins/enterprise_search/public/applications/app_search/utils/results/index.ts

```diff
@@ -5,70 +5,70 @@
  * 2.0.
  */
 
-/* This method flattens fields in documents returned from the ent-search backend.
- * If a field in the document contains "raw" key, it's already flat.
- * If it doesn't, we want to pull properties from it, and move them to the top level.
- * This field is already flat:
- * 'country', { raw: 'United States' }
- * This field is not flat:
- * 'address', {
- *     country: { raw: 'United States' },
- *     city: { raw: 'Los Angeles' }
- * }
- * It will be transformed into:
- * [
- *   ['address.country', { raw: 'United States' }],
- *   ['address.city', { raw: 'Los Angeles' }]
- * ]
- */
-export const flattenDocumentField = (
-  fieldName: string,
-  fieldValue: object
-): Array<[string, object]> => {
-  const flattened: Array<[string, object]> = [];
+function isFieldValueWrapper(object: object): boolean {
+  return (
+    object &&
+    Object.entries(object).reduce((isValueWrapper: boolean, [k, v]) => {
+      if (k !== 'raw' && k !== 'snippet') {
+        return false;
+      }
 
-  if (typeof fieldValue === 'object' && !fieldValue.hasOwnProperty('raw')) {
-    for (const [propName, value] of Object.entries(fieldValue)) {
-      flattenDocumentField(fieldName + '.' + propName, value).map(([flatKey, flatVal]) => {
-        flattened.push([flatKey, flatVal]);
-      });
-    }
-  } else {
-    flattened.push([fieldName, fieldValue]);
+      if (v === null) {
+        return isValueWrapper;
+      }
+
+      return (Array.isArray(v) ? v : []).reduce((isScalar, currentValue) => {
```

Comment:
nit: You can remove a few characters by writing this as ` (Array.isArray(v) || [])`